[FEAT] App Button 추가

devzoe opened the pull request:
## 📌 관련 이슈
<!-- 관련 이슈 번호를 입력해주세요. 예: #123 -->
#12 

## ✨ 작업 내용
<!-- 변경 사항을 간략히 요약해주세요.
   - 무엇을 변경했는지
   - 왜 변경했는지 -->

App Button 을 구현했습니다.
style(primary, secondary)와 size(small, medium, large)를 기준으로 각각의 상태에 따라 backgroundColor, foregroundColor, height를 내부에서 자동으로 처리하도록 MaterialStateProperty.resolveWith 기반으로 구현했습니다.

```dart
// enabled
AppButton(
    label: 'button',
    icon: Icons.camera_alt,
    style: style,
    size: size,
    showText: true,
    showIcon: true,
    onPressed: () {
        debugPrint('Pressed');
    },
)

// disabled
// onPressed에 null을 보내면 disabled
AppButton(
    label: 'button',
    icon: Icons.camera_alt,
    style: style,
    size: size,
    showText: true,
    showIcon: true,
    onPressed: null, 
)
```

## 📸 스크린샷 (선택)
| App Button (primary, secondary) 1 |  App Button (primary, secondary) 2 | 
| ----------------------------------- | -------------------------------------- |
|  ![Screenshot_20250706_160626](https://github.com/user-attachments/assets/888e2d16-b8b9-4330-aa25-6223596b143e) |  ![Screenshot_20250706_160643](https://github.com/user-attachments/assets/01a2d2a0-9781-46cd-b3d3-b608a7abf747) |

## 📚 참고 자료
<!-- 참고한 가이드 문서나 레퍼런스 링크가 있다면 입력해주세요. -->

devzoe committed:
ui: App Button 추가

devzoe committed:
fix: app button content color 수정

devzoe commented:
@generalChang 창호님 App Button에서 secondary-enabled, pressed 일 때 각각 gray100, gray200 컬러인데 화면상으로는 해당 컬러로 안 나오는 이슈가 발생해서.. 혹시 같이 봐주시면 감사하겠습니당!!

generalChang commented:
AppButtonState중에 `pressed`는 버튼과의 상호작용을 통해 지정되는 상태인만큼 외부에서 지정하는 것보단, 내부에서 자체적으로 판단해서 처리하는 것이 좋을 것 같은데 어떻게 생각하시나요?

devzoe commented:
아하 넵넵 state같은 경우는 다시 한번 수정해보도록 할게용! bgColor 이슈도 다시...봐야할 것 같아요!

devzoe committed:
refactor: 버튼 상태 관리 방식을 MaterialState 기반으로 개선
- 기존 AppButtonState enum 기반 상태 분기를 제거하고,
  MaterialStateProperty.resolveWith을 도입하여
  pressed 및 disabled 상태를 자동 감지하도록 변경

devzoe commented:
창호님 다시보니 gray100, gray200 컬러 제대로 나옵니다! 이 이슈는 빼고 봐주셔도 될 것 같아요!

generalChang commented:
린트 경고가 많이 뜨는 것 같습니다 확인 부탁드릴게요..!!

devzoe commented:
헉 넵!!!

generalChang commented on lib/design_system/component/app_button.dart:
```diff
@@ -0,0 +1,100 @@
+
+class AppButton extends StatelessWidget {
+  final String label;
+  final IconData? icon;
```
보통 저희 디자인 시스템에 있는 백터 아이콘 (SVG)를 지정하게 될 것 같은데 SvgGenImage로 받는 것은 어떨까요?

generalChang commented:
AppButtonSize별로 text의 TextStyle과 icon의 size가 분기되어야할 것 같아요 😁

devzoe committed:
refactor: AppButton 상태별 스타일 계산 방식 WidgetStateProperty로 수정
SvgGenImage 타입 아이콘으로 수정
버튼 상태에 따라 아이콘 색상도 자동으로 변경되도록 _buildIcon 메서드에 ColorFilter 적용
버튼 크기에 따라 아이콘 크기 조절하는 _getIconSize 메서드 추가
텍스트 스타일 및 간격 등 size 별 UI 구성 정리

devzoe committed:
Merge branch 'develop' into feat/button-12
